src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -664,7 +665,29 @@ namespace winrt::TerminalApp::implementation
     {
         // Initialize the new tab
 
+        // Try to find a recently used working directory for this profile.
+        Windows::Foundation::Collections::IVectorView<winrt::TerminalApp::Tab> view{ _tabs.GetView() };
+
+        for (auto const& tab : view)
+        {
+            const auto tabImpl = winrt::get_self<winrt::TerminalApp::implementation::Tab>(tab);
+            if (tabImpl)
+            {
+                auto focuesdProfileGuid = tabImpl->GetFocusedProfile();
+                if (profileGuid == focuesdProfileGuid)
+                {
+                    auto workingDirectory = tabImpl->GetActiveTerminalControl().WorkingDirectory();
+                    auto validWorkingDirectory = !workingDirectory.empty() && ::PathFileExists(workingDirectory.data());
```

Comment:
The validation is very important. Without this PowerShell and WSL will refuse to spawn when there's invalid path.

Reply:
Surprisely enough, if you open a PowerShell tab and run `wsl.exe`, the path `\\wsl$\Ubuntu\path` sent by WSL can also be used by PowerShell. It will correctly enter the WSL working directory. I remember seeing this on some article about WSL and Windows filesystem interoperablity. I just didn't not except that it works so seamlessly.

And I think this only works on Windows 2004 and WSL 2. Correct me if I'm wrong.